packages/gestalt/src/Heading.js

```diff
@@ -74,15 +76,16 @@ export default function Heading(props: Props) {
     lgSize && styles[`lgFontSize${SIZE_SCALE[lgSize]}`],
     colors[color],
     overflow === 'breakWord' && typography.breakWord,
-    truncate && typography.truncate
+    truncate && typography.truncate,
+    weight === 'semibold' && typography.fontWeightSemibold
```

Comment:
Do you tend to like to make secondary properties like `fontWeightSemibold` as just overrides to the base class? I will often create two separate classes each that extend the base, then choose which one of those I should use. I'm certain your approach works, but I'm curious what you think, and if you think I'm being too pedantic.

Reply:
@btraut this approach is a bit more atomic, just one property per CSS class and also allows for better re-use across components.